lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -510,23 +519,32 @@ def download_dataset_collection(
 
     @router.post(
         "/api/histories/{history_id}/contents/dataset_collections/{id}/prepare_download",
-        summary="Prepare an short term storage object that the collection will be downloaded to.",
-        responses={
-            200: {
-                "description": "Short term storage reference for async monitoring of this download.",
-            },
-            501: {"description": "Required asynchronous tasks required for this operation not available."},
-        },
+        summary=STORE_DOWNLOAD_DESCRIPTION,
+        responses=STORE_DOWNLOAD_RESPONSES,
+        operation_id="history_contents__prepare_download_dataset_collection",
     )
+    def prepare_collection_download_hc(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        history_id: DecodedDatabaseIdField = HistoryIDPathParam,
```

Comment:
history_id is still unused, don't you think it's better to just have one route that doesn't consume unrelated parameters ?

Reply:
I think both APIs make sense with their respective routes and I find the fact that history_id isn't actually used along the history contents API route an unimportant implementation detail. All the history_contents collection operations could infer the history id and so could all the dataset ones for that matter.

Reply:
Why do we need to have something that looks like a valid history_id (now that we attach a schema to it) to download a collection ? `/api/histories/{history_id}/contents/dataset_collections/{id}/prepare_download"` doesn't even work now ... I had assumed the `/api/histories/{history_id}/contents/` was just for backwards compatibility, so we can drop it from the schema

Reply:
I thought you picked the other route to exclude in your PR - I see. 

These all could work without the history_id:

```
@router.put(
        "/api/histories/{history_id}/contents/{type}s/{id}",
```

```
    @router.delete(
        "/api/histories/{history_id}/contents/{type}s/{id}",
```

```
    @router.post(
        "/api/histories/{history_id}/contents/datasets/{id}/materialize",
```

This API endpoint was modeled after:

```
    @router.get(
        "/api/histories/{history_id}/contents/dataset_collections/{id}/download",
```

which is still there, has a good consistent route, and is fine.

Reply:
True, but for a better UX shouldn't we nudge the API towards `/api/dataset_collections/{id}/download` which is a valid alias for `/api/histories/{history_id}/contents/dataset_collections/{id}/download` ? We obviously have to keep this, but I don't know that we need to document this ?

Reply:
I don't share your preference here... unfortunately I've worked with these APIs long enough that they have warped my brain. ``/api/histories/{history_id}/contents/dataset_collections/{id}/download`` feels right and good to me 😅. If you're operating on a history - that seems like the right place to... find stuff? I've always thought of the other API as the lesser one - and odd little shortcut for certain use cases (I assumed without looking closely it was the one you hid for instance 🤣).

If you insist - I'll let you deprecate and hide these but I don't agree.

Reply:
I think it also makes writing isolated UI components more complicated (or you need to guess what we accept) if you have to carry around a history id even though you're working on a component that has nothing to do with histories. As an example, see 
https://github.com/mvdbeek/galaxy/blob/b107fb7e0deff27ee03bbfffda24dcf0fbd3ddbc/client/src/components/Dataset/services.ts#L68-L79

I don't like to disagree with you, because the few times I do it comes back to bite me, but I think this time I'm willing to risk it ? 💦

Reply:
Boo. This particular call to this API from this super complicated client isn't a reason to drop an API endpoint. But there won't be any negative repercussions. I just need to start reprogramming my brain and while I would argue the API will be a little less intuitive - clearly on balance the API is much more grokable as the result of our efforts.

I will close this PR and I think I've stripped the relevant changes out of the spell check PR?

Reply:
Thank you!